emote/sac.py

```diff
@@ -102,6 +102,7 @@ def __init__(
         roll_length: int = 1,
     ):
         super().__init__()
+        self._order = 1  # this is to ensure that the data_group is prepared beforehand
```

Comment:
Because we have a collector callback, right? Where does the issue occur otherwise?

Reply:
Wondering if we can fix this in the trainer somehow or by making it more closely tied with the loader.

Reply:
The QTarget needs the RL batch at the begin_batch. However, the RL batch is also created at the begin_batch by the BatchSampler. So, an error will occur in case QTarget.begin_batch is called before BatchSampler.begin_batch. So, I set the _order=1 for QTarget to ensure BatchSampler is always called first (its _order is zero by default).

Reply:
Righto! I'll think about this, but we can keep it like it is for now.